service_listener.go

```diff
@@ -182,7 +182,14 @@ func serviceHostname(service *api.Service) (string, error) {
 }
 
 func hostedZoneId(elbApi *elb.ELB, hostname string) (string, error) {
-	elbName := strings.Split(hostname, "-")[0]
+	hostnameSegments := strings.Split(hostname, "-")
+	elbName := hostnameSegments[0]
+	
+	// handle internal load balancer naming
+	if elbName == 'internal' {
+		elbName := hostnameSegments[1]
```

Comment:
First off thanks for the PR! 

That being said, I think this line is incorrect. In golang the `:=` operator is used to instantiate new variables so the scope of this variable lives inside the `if` statement. It seems you just want to reassign the variable so in that case you just need the `=` operator.